src/store.c

```diff
@@ -269,15 +310,14 @@ static bool writer_open(
     }
 
     store->head = store->vma;
-    store->vals = (void *) ((uintptr_t) store->vma + sizeof(struct header));
     store->end = (void *) ((uintptr_t) store->vma + store->vma_len);
 
     *store->head = (struct header) {
         .magic = magic,
         .version = version,
         .ts = ts,
         .quant = quant,
-        .vals_off = sizeof(struct header),
+        .index_a_off = sizeof(struct header),
```

Comment:
Since the indexes are of fixed size and you can know them right now. You could technically set the offset and the pointers for both indexes right now. This centralises most of the offset calculations in one place which would simplify the rest of the code.

Reply:
fixorzed